### DISCUSSION TITLE
task list with bold don't work

### DISCUSSION THREAD

**Discussion by tsohst** [requarks/wiki#3684] [Error / Bug Report]
Using wikijs 2.3.81
### Actual behavior
`- [x] **test**: 1234` in markdown site creates the following string -> **test** `**test**: 1234`

### Expected behavior
- [x] **test**: 1234

### Steps to reproduce the behavior
Create `- [x] **test**: 1234` in markdown site and create `**test**: 1234`
It's a problem when using together with task list.

**Comment by bmann** [comment]
Any sort of formatting -- including markdown links `[links](/path/to/page)` or bare links `https://example.com` don't work with task lists.

**Comment by bmann** [comment]
Duplicate of https://github.com/Requarks/wiki/issues/1856, https://github.com/Requarks/wiki/issues/1860

**Comment by GuillaumeHullin** [comment]
Hi, I'm also in that situation... it seems to be a problem with the 2nd HTML rendering... if I understand correctly, the preview panel is only the first HTML rendering from Markdown but there is also a second one after.

My dirty workaround right now is to insert this code in the Head HTML injection via the Admin panel.

```html
<script type="text/javascript">
    window.addEventListener("load", function(){
        var render = document.getElementsByClassName("contents")[0];
        var lbl = render.getElementsByClassName("task-list-item-label");
        for (var i = 0; i < lbl.length; i++) {
            lbl[i].innerHTML = lbl[i].innerHTML.replace(/.*\*.+\*/,'');
        }
    });
</script>
```

I will try to look a bit more in the code of Wiki.js but if anybody have a clue where I should look, it would be useful.

**Comment by GuillaumeHullin** [comment]
Ok I think I found a solution to this bug... but as I'm not confortable with docker containers I don't want to create a Pull Request... but here is what I did:

I modified **/server/modules/rendering/markdown-tasklists/renderer.js**

```bash
docker exec -it wiki vi server/modules/rendering/markdown-tasklists/renderer.js
```

I replaced
```js
md.use(mdTaskLists, {label: true, labelAfter: true})
```

with:
```js
 md.use(mdTaskLists, {label: true, labelAfter: false})
```

And now, after a re-rendering of the test page, all the formatting worked (bold, italics, links, keyboard key etc...)

![Screenshot 2022-02-12 at 23 06 40](https://user-images.githubusercontent.com/4108400/153730129-0f8ecf20-759c-43d9-87e7-96b5ebcbaa42.png)

**Before**
![Screenshot 2022-02-12 at 23 09 31](https://user-images.githubusercontent.com/4108400/153730134-cb820707-b1f4-49ac-9bcd-6f418042464b.png)

**After**
![Screenshot 2022-02-12 at 23 10 36](https://user-images.githubusercontent.com/4108400/153730142-c01b9a4b-de11-4b2e-9fcd-f54dfc5dcd27.png)


I'm not exactly sure what it change but it seems to change the position of the label html tag... https://www.npmjs.com/package/markdown-it-task-lists

That might interest you @NGPixel ;)

**Reply by NGPixel** [reply]
That was actually already fixed in the editor, but for some reason the fix wasn't applied to the server as well. I just made the change to the server renderer to match: afafb4f4e0d49d76d7c050fedafa0678e121739a

**Reply by GuillaumeHullin** [reply]
Awesome. Thanks for the quick fix.
I have a "stupid" question (just I don't fully understand yet how docker work)... how do I get this commit to be applied on my container... should I redownload the image... or else?

**Reply by NGPixel** [reply]
The release is currently building, so wait about 40 mins for the images to become available. Then upgrade using the standard [upgrade](https://docs.requarks.io/install/upgrade) instructions. Check https://github.com/Requarks/wiki/releases/tag/v2.5.275 to see when it's live (no longer says "Prerelease").

However, all commits have a "dev" image built, so you could also use `ghcr.io/requarks/wiki:canary-2.5.0-dev.12` right away, but I would wait for the final release instead...

**Reply by GuillaumeHullin** [reply]
Thanks for taking the time answering. I'll wait tomorrow morning and follow the procedure.
